robottelo/ui/base.py

```diff
@@ -82,8 +82,14 @@ def navigate_to_entity(self):
         """Perform navigation to main page for specific entity"""
         raise NotImplementedError('Subclasses must implement navigator method')
 
-    def search(self, element_name):
-        """Uses the search box to locate an element from a list of elements."""
+    def search(self, element):
+        """Uses the search box to locate an element from a list of elements.
+
+        :param element: either element name or a tuple, containing element name
+        as a first element and all the rest variables required for element
+        locator.
+        """
+        element_name = element[0] if isinstance(element, tuple) else element
```

Comment:
Currently we have at least 2 entity locators dependent on multiple fields - bookmarks and docker tags. Bookmarks can have the same name within different controllers, docker tags - same name within different repositories and products. It doesn't sound good to me to copy-paste entire Base.search() method for them with the only 1 line change - in `self.find_element((strategy, value % element))`, so i decided to extend base search to accept tuples too. This way all the regular entities remain untouched and for bookmarks or docker tags we can just pass the tuple like this:
```self.dockertag.search(('latest', pr.name, repo.name))```